clippy_lints/src/needless_braces_on_range_literal.rs

```diff
@@ -0,0 +1,67 @@
+//! Checks for braces on literals in range statements
+//!
+//! For example, the lint would catch
+//!
+//! ```rust
+//! for i in (0)..10 {
+//!   println!("{i}");
+//! }
+//! ```
+//!
+//! And suggest this:
+//!
+//! ```rust
+//! for i in 0..10 {
+//!   println!("{i}");
+//! }
+//! ```
+//!
+//! This lint is **warn** by default.
+
+use clippy_utils::{diagnostics::span_lint_and_then, source::snippet_opt};
+use rustc_ast::ast::{Expr, ExprKind};
+use rustc_errors::Applicability;
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+
+declare_clippy_lint! {
+  /// ### What it does
+  /// The lint checks for braces on literals in range statements that are
+  /// superflous.
+  ///
+  /// ### Why is this bad?
+  /// Having superflous braces makes the code less legible as the impose an
+  /// overhead when reading.
+
```

Comment:
It looks like you took the examples out of here and put them at the top of the file. They need to be within this macro for the docs to appear on the [lint list](https://rust-lang.github.io/rust-clippy/master).

Reply:
Implemented in 8185ecd7